scripts/common/src/lib.rs

```diff
@@ -0,0 +1,233 @@
+}
+
+pub async fn load_stylus_config() -> Result<StylusConfig> {
+    let project_root = find_parent_project_root(None)?;
+    let stylus_config_path = project_root.join(STYLUS_CONFIG_FILENAME);
+    if !fs::try_exists(&stylus_config_path).await? {
+        bail!(
+            "{} not found at {}",
+            STYLUS_CONFIG_FILENAME,
+            project_root.display()
+        );
+    }
+    let stylus_config_str = fs::read_to_string(&stylus_config_path).await?;
+    Ok(toml::from_str(&stylus_config_str)?)
+}
+
+pub async fn load_network_config_for(network: &str) -> Result<NetworkConfig> {
+    let mut stylus_config: StylusConfig = load_stylus_config().await?;
+
+    stylus_config
+        .networks
+        // NOTE: `.remove`-ing instead of `.get`-ing to avoid `.clone()`-ing later
+        .remove(&network.to_lowercase())
+        .ok_or_eyre(format!("No configuration for network {}", network))
+}
+
+pub async fn load_config_for(network: &str) -> Result<Config> {
+    let NetworkConfig {
+        rpc_url,
+        private_key_source,
+        additional_variables,
+    } = load_network_config_for(network).await?;
+
+    let private_key = private_key_source.load()?;
+
+    // Prepare client
+    let provider = Provider::<Http>::try_from(rpc_url.clone())?;
+    let wallet = LocalWallet::from_str(&private_key)?;
+    let chain_id = provider.get_chainid().await?.as_u64();
+    let client = Arc::new(SignerMiddleware::new(
+        provider,
+        wallet.clone().with_chain_id(chain_id),
+    ));
+
+    Ok(Config {
+        client,
+        additional_variables,
+    })
+}
+
+/// Reads and trims a line from a filepath
+pub fn read_secret_from_file(fpath: impl AsRef<Path>) -> eyre::Result<String> {
+    let f = std::fs::File::open(fpath)?;
+    let mut buf_reader = BufReader::new(f);
+    let mut secret = String::new();
+    buf_reader.read_line(&mut secret)?;
+    Ok(secret.trim().to_string())
+}
+
+pub fn find_parent_project_root(start_from: Option<PathBuf>) -> Result<PathBuf> {
+    let start_from = start_from.unwrap_or(env::current_dir()?);
+
+    //  NOTE: search upwards for `.git`
+    cargo_stylus::util::discover_project_root_from_path(start_from)?
+        .ok_or_eyre("Could not find project root")
+}
+
+pub fn move_to_parent_project_root() -> Result<()> {
+    let parent_project_root = &find_parent_project_root(None)?;
+
+    env::set_current_dir(parent_project_root)?;
+    println!("Set cwd to {}", parent_project_root.display());
+
+    Ok(())
+}
+
+pub fn make_absolute_relative_to(
+    path: impl AsRef<Path>,
+    relative_to: impl AsRef<Path>,
+) -> Result<PathBuf> {
+    let mut path: PathBuf = path.as_ref().to_path_buf();
+    let relative_to = relative_to.as_ref();
+
+    if !path.is_absolute() {
+        path = relative_to.join(path);
+    }
+
+    path.canonicalize()
+        .wrap_err(format!("Could not canonicalize {}", path.display()))
+}
+
+pub async fn deploy_on(network: &str) -> Result<Address> {
```

Comment:
I'd rename this function to `deploy` because if we decide to extend it with more parameters, we'd have to rename it anyways because the name won't sound right.